### PR TITLE
NAC-285: Akita Prometheus Exporter (upstart + env)

### PR DESCRIPTION
**Pull request by mbowen**

### CODE REVIEW

**Comment by benjaminkreen** [comment]
@mbowen can you kinda give me a run down about whats going on? There's a lot of ruby code for a salt change here

File: akita/opt/prometheus/prometheus-exporter/akita_collector.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class AkitaCollector < PrometheusExporter::Server::CollectorBase
```
Comment (benjaminkreen) [inline akita/opt/prometheus/prometheus-exporter/akita_collector.rb]: How does this class work with the rest of the prometheus exporter gem?

Comment (mbowen) [reply akita/opt/prometheus/prometheus-exporter/akita_collector.rb]: sorry for not responding sooner. the collector defines the specific metrics, in this case, two counters. one to capture all accounts created, the other just verified accounts. prometheus polls metrics and records timestamp which each collection. we're able to run prometheus query over an interval to get delta (ex: how many accounts created yesterday).

Comment (mbowen) [reply akita/opt/prometheus/prometheus-exporter/akita_collector.rb]: the mock in akita is basically the custom collector code, but replace prometheus counters with simple integers. could this be dryer? maybe. would have been easier if they could run under the same version of ruby. so, collector code is defined in salt (akita formula).

File: akita/opt/prometheus/prometheus-exporter/akita_collector.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    @verified_users_cnt = PrometheusExporter::Metric::Counter.new('verified_users_count', 'Total # of verified accounts')
+    @all_users_cnt = PrometheusExporter::Metric::Counter.new('all_users_count', 'Total # of accounts created (verified & unverified)')
+    @mutex = Mutex.new
+  end
+
+  def process(str)
+    obj = JSON.parse(str)
+    @mutex.synchronize do
+      if verified_cnt = obj["verified_users_count"]
+        @verified_users_cnt.observe(verified_cnt)
```
Comment (sgatchell) [inline akita/opt/prometheus/prometheus-exporter/akita_collector.rb]: Do counters have an `observe` method?  I only saw it on summary/histogram:  https://github.com/prometheus/client_ruby/search?q=observe&unscoped_q=observe

Comment (sgatchell) [reply akita/opt/prometheus/prometheus-exporter/akita_collector.rb]: oops wrong library, sorry

Comment (sgatchell) [reply akita/opt/prometheus/prometheus-exporter/akita_collector.rb]: got it.  `observe` just wraps `increment` in this case: https://github.com/discourse/prometheus_exporter/blob/master/lib/prometheus_exporter/metric/counter.rb

Comment (mbowen) [reply akita/opt/prometheus/prometheus-exporter/akita_collector.rb]: cool. thanks for digging into this. akita formula with prometheus exporter (metrics) is still in the incubator stage. there were some challenges with rigging this up. exporter uses a newer version of ruby (2.3+) than akita (2.2). we initially tried plumbing metrics using official prometheus for ruby client, but had some problems with the puma workers. documented more in NAC-285. ben, chris, and i are still iterating on this.

File: akita/prometheus-exporter.sls [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+{% set app_home  = user_home ~ '/' ~ app %}
+
+{{ app_home }}:
+  file.directory:
+    - user: {{ user }} 
+    - group: {{ user }}
+    - require:
+      - user: {{ user }}
+
+{{ user }}-ruby-{{ ruby_ver }}:
```
Comment (benjaminkreen) [inline akita/prometheus-exporter.sls]: would https://github.com/PLOS/molten/blob/dev/states/lib/ruby.sls#L22 be a handy macro here?

**Comment by benjaminkreen** [comment]
@wryfi to refactor in https://jira.plos.org/jira/browse/SRE-256

**Pull request merged**